File: src/BooksReader.Web/ClientApp/src/app/modules/core/models/api-contracts/book-edit-info.ts
import { Book } from './entities/book';
import { BookChapter } from './entities/book-chapter';
import { PersonalPage } from './entities';
import { BookPrice } from './entities/book-price';

export interface BookEditInfo {
    book: Book;
    bookPage: PersonalPage;
    chapters: BookChapter[];
    prices: BookPrice[];
}

File: src/BooksReader.Core/Models/BookEditInfo.cs
using System.Collections.Generic;
using BooksReader.Core.Entities;

namespace BooksReader.Core.Models
{
    public class BookEditInfo
    {
        public Book Book { get; set; }
        public PersonalPage BookPage{ get; set; }
        public IEnumerable<BookChapter> Chapters { get; set; }
        public IEnumerable<BookPrice> Prices { get; set; }
    }
}
